### ISSUE TITLE
Alpha 0.7v2 error

### ISSUE THREAD

**Issue by hejman** [Diblo/KODI-Popcorn-Time#63]
When pressing "Clear Cache" on Mac

22:40:50 T:123145302839296   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.SyntaxError'>
                                            Error Contents: ('invalid syntax', ('/Users/kno/Library/Application Support/Kodi/addons/plugin.video.kodipopcorntime/resources/lib/kodipopcorntime/torrent.py', 386, 46, "            with open(self._path, 'r') as src, open(self._tmppath, 'w') as dst:\n"))
                                            Traceback (most recent call last):
                                              File "/Users/kno/Library/Application Support/Kodi/addons/plugin.video.kodipopcorntime/addon.py", line 5, in <module>
                                                from kodipopcorntime import main
                                              File "/Users/kno/Library/Application Support/Kodi/addons/plugin.video.kodipopcorntime/resources/lib/kodipopcorntime/main.py", line 9, in <module>
                                                from kodipopcorntime.torrent import TorrentPlayer
                                            SyntaxError: ('invalid syntax', ('/Users/kno/Library/Application Support/Kodi/addons/plugin.video.kodipopcorntime/resources/lib/kodipopcorntime/torrent.py', 386, 46, "            with open(self._path, 'r') as src, open(self._tmppath, 'w') as dst:\n"))
                                            -->End of Python script error report<--

**Comment by Diblo** [comment]
Look for a opdate later today :+:

**Comment by Diblo** [comment]
Try again https://github.com/Diblo/KODI-Popcorn-Time/wiki/Test-releases#download-1

**Comment by hejman** [comment]
I get the same error message.
```python
17:55:22 T:123145302839296   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.SyntaxError'>
                                            Error Contents: ('invalid syntax', ('/Users/kno/Library/Application Support/Kodi/addons/plugin.video.kodipopcorntime/resources/lib/kodipopcorntime/torrent.py', 386, 46, "            with open(self._path, 'r') as src, open(self._tmppath, 'w') as dst:\n"))
                                            Traceback (most recent call last):
                                              File "/Users/kno/Library/Application Support/Kodi/addons/plugin.video.kodipopcorntime/addon.py", line 5, in <module>
                                                from kodipopcorntime import main
                                              File "/Users/kno/Library/Application Support/Kodi/addons/plugin.video.kodipopcorntime/resources/lib/kodipopcorntime/main.py", line 9, in <module>
                                                from kodipopcorntime.torrent import TorrentPlayer
                                            SyntaxError: ('invalid syntax', ('/Users/kno/Library/Application Support/Kodi/addons/plugin.video.kodipopcorntime/resources/lib/kodipopcorntime/torrent.py', 386, 46, "            with open(self._path, 'r') as src, open(self._tmppath, 'w') as dst:\n"))
                                            -->End of Python script error
```

**Comment by hejman** [comment]
We still have a problem, the plugin is looking for torrent2http in darwin_x86 which doesn't exist.

@Diblo

**Comment by hejman** [comment]
After renaming the /bin/darwin_x64 torrent2http folder to darwin_x86 i managed to play a movie, however Kodi will not shut down, see below.
(this is another issue)

```Python
18:25:01 T:140735219187712   DEBUG: ADDON: Stopping service addons.
18:25:01 T:123145310507008    NONE:   DEBUG: [plugin.video.kodipopcorntime] (Torrent) (torrent2http) Removing the torrent
18:25:01 T:123145310507008    NONE:   DEBUG: [plugin.video.kodipopcorntime] (Torrent) (torrent2http) Waiting for files to be removed
18:25:01 T:123145310507008    NONE:   DEBUG: [plugin.video.kodipopcorntime] (Torrent) (torrent2http) Aborting the session
18:25:06 T:140735219187712   ERROR: CPythonInvoker(3, /Users/kno/Library/Application Support/Kodi/addons/plugin.video.kodipopcorntime/addon.py): script didn't stop in 5 seconds - let's kill it
18:25:06 T:123145315336192    NONE:   DEBUG: [plugin.video.kodipopcorntime] (Torrent) (torrent2http) Bye bye
18:25:06 T:123145309970432    NONE:    INFO: [plugin.video.kodipopcorntime] (Torrent) torrent2http successfully shutdown
18:25:06 T:123145309970432    INFO: CPythonInvoker(17, /Users/kno/Library/Application Support/Kodi/addons/plugin.video.kodipopcorntime/addon.py): script successfully run
18:25:06 T:123145309970432    INFO: CPythonInvoker(17, /Users/kno/Library/Application Support/Kodi/addons/plugin.video.kodipopcorntime/addon.py): waiting on thread 123145315336192
18:25:07 T:123145306673152    NONE: WARNING: [plugin.video.kodipopcorntime] (Torrent) An error occurred while shutting down torrent2http
18:25:07 T:123145306673152    NONE: WARNING: [plugin.video.kodipopcorntime] (Torrent) Killing torrent2http
18:25:07 T:123145306673152    INFO: CPythonInvoker(3, /Users/kno/Library/Application Support/Kodi/addons/plugin.video.kodipopcorntime/addon.py): script successfully run
18:25:07 T:123145306673152    INFO: CPythonInvoker(3, /Users/kno/Library/Application Support/Kodi/addons/plugin.video.kodipopcorntime/addon.py): waiting on thread 123145316409344
18:25:08 T:123145310507008    NONE:   DEBUG: [plugin.video.kodipopcorntime] (Torrent) (torrent2http) Bye bye
18:25:08 T:123145305063424    NONE:    INFO: [plugin.video.kodipopcorntime] (Torrent) torrent2http successfully shutdown
18:25:08 T:123145305063424    INFO: CPythonInvoker(19, /Users/kno/Library/Application Support/Kodi/addons/plugin.video.kodipopcorntime/addon.py): script successfully run
18:25:08 T:123145305063424    INFO: CPythonInvoker(19, /Users/kno/Library/Application Support/Kodi/addons/plugin.video.kodipopcorntime/addon.py): waiting on thread 123145310507008
18:26:00 T:123145311580160 WARNING: DARWINOSX: underflow (0 vs 4096 bytes)```

**Comment by Diblo** [comment]
@hejman 
Can you try again https://drive.google.com/file/d/0B3fNnjj64nUpdDN3WXBFdDZEd0U/view?usp=sharing.
You must rename darwin_x64 to darwin_x86. (If it works I will add a solution.)

**Comment by Diblo** [comment]
At one point, our link to the Alpha release stopped to link to the latest zip file. As result we have to ask you to reinstall the the addon so we and you can get a properly test result.
https://drive.google.com/file/d/0B3fNnjj64nUpdnZXM2dYb05NUzg/view?usp=sharing

**Comment by hejman** [comment]
Still freezing
``Python
18:00:37 T:140735219187712  NOTICE: Storing total System Uptime
18:00:37 T:140735219187712  NOTICE: Saving settings
18:00:37 T:140735219187712  NOTICE: stop all
18:00:37 T:123145318019072   DEBUG: Thread JobWorker 123145318019072 terminating (autodelete)
18:00:37 T:140735219187712  NOTICE: stop player
18:00:37 T:140735219187712  NOTICE: CDVDPlayer::CloseFile()
18:00:37 T:140735219187712  NOTICE: DVDPlayer: waiting for threads to exit
18:00:37 T:123145304526848  NOTICE: CDVDPlayer::OnExit()
18:00:37 T:123145304526848  NOTICE: Closing stream player 1
18:00:37 T:123145304526848  NOTICE: Waiting for audio thread to exit
18:00:37 T:123145322311680  NOTICE: thread end: CDVDPlayerAudio::OnExit()
18:00:37 T:123145304526848  NOTICE: Closing audio device
18:00:37 T:123145322311680   DEBUG: Thread DVDPlayerAudio 123145322311680 terminating
18:00:37 T:123145307209728   DEBUG: CActiveAE::DiscardStream - audio stream deleted
18:00:37 T:123145304526848  NOTICE: Deleting audio codec
18:00:37 T:123145304526848  NOTICE: Closing stream player 2
18:00:37 T:123145304526848  NOTICE: waiting for video thread to exit
18:00:37 T:123145307209728   DEBUG: CActiveAE::ClearDiscardedBuffers - buffer pool deleted
18:00:37 T:123145307746304   DEBUG: Previous line repeats 1 times.
18:00:37 T:123145307746304    INFO: CActiveAESink::OpenSink - initialize sink
18:00:37 T:123145321775104  NOTICE: thread end: video_thread
18:00:37 T:123145321775104   DEBUG: Thread DVDPlayerVideo 123145321775104 terminating
18:00:37 T:123145304526848  NOTICE: deleting video codec
18:00:37 T:123145305063424   DEBUG: Thread FileCache 123145305063424 terminating
18:00:37 T:123145304526848   DEBUG: OnPlayBackStopped: play state was 2, starting 0
18:00:37 T:123145304526848   DEBUG: CAnnouncementManager - Announcement: OnStop from xbmc
18:00:37 T:123145304526848   DEBUG: GOT ANNOUNCEMENT, type: 1, from xbmc, message OnStop
18:00:37 T:123145304526848   DEBUG: Thread DVDPlayer 123145304526848 terminating
18:00:37 T:140735219187712  NOTICE: DVDPlayer: finished waiting
18:00:37 T:140735219187712   DEBUG: LinuxRendererGL: Cleaning up GL resources
18:00:37 T:140735219187712  NOTICE: CDVDPlayer::CloseFile()
18:00:37 T:140735219187712  NOTICE: DVDPlayer: waiting for threads to exit
18:00:37 T:140735219187712  NOTICE: DVDPlayer: finished waiting
18:00:37 T:140735219187712   DEBUG: LinuxRendererGL: Cleaning up GL resources
18:00:37 T:123145317482496   DEBUG: [plugin.video.kodipopcorntime] (Torrent) (torrent2http) [3] Closing Terminator Genisys (2015)/Terminator.Genisys.2015.720p.BluRay.x264.YIFY.mp4...
18:00:37 T:140735219187712    INFO: stopping PVRManager
18:00:37 T:140735219187712   DEBUG: NetworkMessage - Signaling network services to stop
18:00:37 T:140735219187712  NOTICE: ES: Stopping event server
18:00:37 T:140735219187712   DEBUG: NetworkMessage - Waiting for network services to stop
18:00:37 T:140735219187712  NOTICE: stopping upnp
18:00:37 T:123145308819456   DEBUG: [plugin.video.kodipopcorntime] (Torrent Player) The playback has stop
18:00:37 T:123145308819456   DEBUG: [plugin.video.kodipopcorntime] (Torrent) Shutting down log pipe
18:00:37 T:123145308819456   DEBUG: [plugin.video.kodipopcorntime] (Torrent) Shutting down torrent2http
18:00:37 T:123145308819456   DEBUG: [plugin.video.kodipopcorntime] (URL) Headers: {'Accept-Encoding': 'gzip', 'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/30.0.1599.66 Safari/537.36'}
18:00:37 T:123145308819456   DEBUG: [plugin.video.kodipopcorntime] (URL) Trying to obtaining data from http://127.0.0.1:60710/shutdown
18:00:37 T:123145317482496   DEBUG: [plugin.video.kodipopcorntime] (Torrent) (torrent2http) Stopping torrent2http...
18:00:37 T:123145317482496   DEBUG: [plugin.video.kodipopcorntime] (Torrent) (torrent2http) Logging finished
18:00:38 T:123145308819456    INFO: [plugin.video.kodipopcorntime] (Torrent) torrent2http successfully shutdown
18:00:38 T:123145308819456    INFO: CPythonInvoker(17, /Users/kno/Library/Application Support/Kodi/addons/plugin.video.kodipopcorntime/addon.py): script successfully run
18:00:38 T:123145307746304   DEBUG: CCoreAudioStream::Close: Restoring original physical format for stream 0x0033. ([lpcm] Mixable Interleaved 2 Channel 32-bit Floating Point LE (44100Hz))
18:00:38 T:140735219187712  NOTICE: stopping zeroconf publishing
18:00:38 T:123145308819456 WARNING: CPythonInvoker(17, /Users/kno/Library/Application Support/Kodi/addons/plugin.video.kodipopcorntime/addon.py): the python script "/Users/kno/Library/Application Support/Kodi/addons/plugin.video.kodipopcorntime/addon.py" has left several classes in memory that we couldn't clean up. The classes include: N9XBMCAddon7xbmcgui12ControlLabelE,N9XBMCAddon7xbmcgui12ControlImageE,N9XBMCAddon7xbmcgui6WindowE,N14PythonBindings30XBMCAddon_xbmc_Player_DirectorE,N9XBMCAddon7xbmcgui14DialogProgressE
18:00:38 T:123145308819456    INFO: Python script stopped
18:00:38 T:123145308819456   DEBUG: Thread LanguageInvoker 123145308819456 terminating
18:00:38 T:123145307746304   DEBUG: CCoreAudioStream::SetPhysicalFormat: Physical format for stream 0x0033. now active ([lpcm] Mixable Interleaved 2 Channel 32-bit Floating Point LE (44100Hz))
18:00:38 T:123145307746304   DEBUG: CCoreAudioStream::Close: Closed stream 0x0033.
18:00:38 T:123145307746304   DEBUG: CActiveAESink::OpenSink - trying to open device DARWINOSX:default
18:00:38 T:123145307746304  NOTICE: virtual bool CAESinkDARWINOSX::Initialize(AEAudioFormat &, std::string &): Opening default device Built-in Output
18:00:38 T:123145307746304   DEBUG: FindSuitableFormatForStream: Finding stream for format AE_FMT_FLOAT
18:00:38 T:123145307746304   DEBUG: FindSuitableFormatForStream: Physical Format: [lpcm] Mixable Interleaved 2 Channel 24-bit Signed Integer LE (96000Hz) rated 929.129272
18:00:38 T:123145307746304   DEBUG: FindSuitableFormatForStream: Physical Format: [lpcm] Mixable Interleaved 2 Channel 24-bit Signed Integer LE (88200Hz) rated 1106.000000
18:00:38 T:123145307746304   DEBUG: FindSuitableFormatForStream: Physical Format: [lpcm] Mixable Interleaved 2 Channel 24-bit Signed Integer LE (48000Hz) rated 1419.564697
18:00:38 T:123145307746304   DEBUG: FindSuitableFormatForStream: Physical Format: [lpcm] Mixable Interleaved 2 Channel 24-bit Signed Integer LE (44100Hz) rated 1508.000000
18:00:38 T:123145307746304   DEBUG: FindSuitableFormatForStream: Physical Format: [lpcm] Mixable Interleaved 2 Channel 20-bit Signed Integer LE (96000Hz) rated 928.129272
18:00:38 T:123145307746304   DEBUG: FindSuitableFormatForStream: Physical Format: [lpcm] Mixable Interleaved 2 Channel 20-bit Signed Integer LE (88200Hz) rated 1105.000000
18:00:38 T:123145307746304   DEBUG: FindSuitableFormatForStream: Physical Format: [lpcm] Mixable Interleaved 2 Channel 20-bit Signed Integer LE (48000Hz) rated 1418.564697
18:00:38 T:123145307746304   DEBUG: FindSuitableFormatForStream: Physical Format: [lpcm] Mixable Interleaved 2 Channel 20-bit Signed Integer LE (44100Hz) rated 1507.000000
18:00:38 T:123145307746304   DEBUG: FindSuitableFormatForStream: Physical Format: [lpcm] Mixable Interleaved 2 Channel 16-bit Signed Integer LE (96000Hz) rated 928.129272
18:00:38 T:123145307746304   DEBUG: FindSuitableFormatForStream: Physical Format: [lpcm] Mixable Interleaved 2 Channel 16-bit Signed Integer LE (88200Hz) rated 1105.000000
18:00:38 T:123145307746304   DEBUG: FindSuitableFormatForStream: Physical Format: [lpcm] Mixable Interleaved 2 Channel 16-bit Signed Integer LE (48000Hz) rated 1418.564697
18:00:38 T:123145307746304   DEBUG: FindSuitableFormatForStream: Physical Format: [lpcm] Mixable Interleaved 2 Channel 16-bit Signed Integer LE (44100Hz) rated 1507.000000
18:00:38 T:123145307746304   DEBUG: FindSuitableFormatForStream: Physical Format: [lpcm] Mixable Interleaved 2 Channel 32-bit Floating Point LE (96000Hz) rated 930.129272
18:00:38 T:123145307746304   DEBUG: FindSuitableFormatForStream: Physical Format: [lpcm] Mixable Interleaved 2 Channel 32-bit Floating Point LE (88200Hz) rated 1107.000000
18:00:38 T:123145307746304   DEBUG: FindSuitableFormatForStream: Physical Format: [lpcm] Mixable Interleaved 2 Channel 32-bit Floating Point LE (48000Hz) rated 1420.564697
18:00:38 T:123145307746304   DEBUG: FindSuitableFormatForStream: Physical Format: [lpcm] Mixable Interleaved 2 Channel 32-bit Floating Point LE (44100Hz) rated 1509.000000
18:00:38 T:123145307746304   DEBUG: Initialize: Selected stream[0] - id: 0x0033, Physical Format: [lpcm] Mixable Interleaved 2 Channel 32-bit Floating Point LE (44100Hz)
18:00:38 T:123145307746304   DEBUG: CCoreAudioDevice::SupportsMixing: Device mixing support : 'No'.
18:00:38 T:123145307746304   DEBUG: CCoreAudioStream::Open: Opened stream 0x0033.
18:00:38 T:123145307746304   DEBUG: Initialize: Previous Virtual Format: [lpcm] Mixable Interleaved 2 Channel 32-bit Floating Point LE (44100Hz)
18:00:38 T:123145307746304   DEBUG: Initialize: Previous Physical Format: [lpcm] Mixable Interleaved 2 Channel 32-bit Floating Point LE (44100Hz)
18:00:38 T:123145307746304   DEBUG: CCoreAudioStream::SetPhysicalFormat: Physical format for stream 0x0033. now active ([lpcm] Mixable Interleaved 2 Channel 32-bit Floating Point LE (44100Hz))
18:00:38 T:123145307746304   DEBUG: Initialize: New Virtual Format: [lpcm] Mixable Interleaved 2 Channel 32-bit Floating Point LE (44100Hz)
18:00:38 T:123145307746304   DEBUG: Initialize: New Physical Format: [lpcm] Mixable Interleaved 2 Channel 32-bit Floating Point LE (44100Hz)
18:00:38 T:123145307746304   DEBUG: GetAEChannelMap Engine requests layout FL,FR
18:00:38 T:123145307746304   DEBUG: GetAEChannelMap trying to map to stereo layout: [Left] [Right]
18:00:38 T:123145307746304   DEBUG: GetAEChannelMap mapped channels to layout FL,FR
18:00:38 T:123145307746304   DEBUG: Initialize: using buffer size: 16384 (0.046440 ms)
18:00:38 T:123145307746304   DEBUG: CActiveAESink::OpenSink - DARWINOSX Initialized:
18:00:38 T:123145307746304   DEBUG:   Output Device : Default
18:00:38 T:123145307746304   DEBUG:   Sample Rate   : 44100
18:00:38 T:123145307746304   DEBUG:   Sample Format : AE_FMT_FLOAT
18:00:38 T:123145307746304   DEBUG:   Channel Count : 2
18:00:38 T:123145307746304   DEBUG:   Channel Layout: FL,FR
18:00:38 T:123145307746304   DEBUG:   Frames        : 512
18:00:38 T:123145307746304   DEBUG:   Frame Samples : 1024
18:00:38 T:123145307746304   DEBUG:   Frame Size    : 8
18:00:38 T:123145307209728   DEBUG: CActiveAE::ClearDiscardedBuffers - buffer pool deleted
18:00:38 T:123145314263040  NOTICE: ES: UDP Event server stopped
18:00:38 T:123145314799616   DEBUG: Thread TCPServer 123145314799616 terminating
18:00:38 T:123145314263040   DEBUG: Thread EventServer 123145314263040 terminating
18:00:38 T:140735219187712  NOTICE: stop dvd detect media
18:00:38 T:123145309433856   DEBUG: Thread DetectDVDMedia 123145309433856 terminating
18:00:38 T:123145309356032   DEBUG: Thread PeripBusCEC 123145309356032 terminating
18:00:38 T:140735219187712  NOTICE: stop sap announcement listener
18:00:38 T:140735219187712  NOTICE: clean cached files!
18:00:38 T:140735219187712  NOTICE: unload skin
18:00:38 T:140735219187712    INFO: Unloading old skin ...
18:00:38 T:140735219187712   DEBUG: ------ Window Deinit () ------
18:00:38 T:140735219187712   DEBUG: ------ Window Deinit (Home.xml) ------
18:00:38 T:140735219187712   DEBUG: ------ Window Deinit (Pointer.xml) ------
18:00:38 T:140735219187712   DEBUG: ------ Window Deinit (DialogButtonMenu.xml) ------
18:00:38 T:140735219187712   DEBUG: Cleanup - Closed bundle
18:00:38 T:140735219187712   DEBUG: Infobool 'isempty(listitem.property(addon.broken))' still used by 2 instances
18:00:38 T:140735219187712   DEBUG: Infobool '!isempty(listitem.property(addon.broken))' still used by 2 instances
18:00:38 T:140735219187712   DEBUG: Infobool 'player.showcodec' still used by 2 instances
18:00:38 T:140735219187712   DEBUG: ADDON: Stopping service addons.
18:00:43 T:140735219187712   ERROR: CPythonInvoker(3, /Users/kno/Library/Application Support/Kodi/addons/plugin.video.kodipopcorntime/addon.py): script didn't stop in 5 seconds - let's kill it
18:01:38 T:123145308819456 WARNING: DARWINOSX: underflow (0 vs 4096 bytes)
18:01:48 T:123145307746304   DEBUG: CCoreAudioStream::Close: Restoring original physical format for stream 0x0033. ([lpcm] Mixable Interleaved 2 Channel 32-bit Floating Point LE (44100Hz))
18:01:48 T:123145307746304   DEBUG: CCoreAudioStream::SetPhysicalFormat: Physical format for stream 0x0033. now active ([lpcm] Mixable Interleaved 2 Channel 32-bit Floating Point LE (44100Hz))
18:01:48 T:123145307746304   DEBUG: CCoreAudioStream::Close: Closed stream 0x0033.```

**Comment by hejman** [comment]
Any news on this @Diblo ?
